### PR TITLE
Limit root_caches to only contain target root dependencies. 

### PR DESCRIPTION
**Pull request by Ladsgroup**
Bug: T214841

### CODE REVIEW

**Comment by halfak** [comment]
This doesn't quite do what it needs to.  The root_caches need to include the injected_cache when extract is called.  So if someone specifies a root datasource, it is then used during extraction.

File: ores/scoring_systems/tests/test_single_thread.py [hunk 1]
```diff
@@ -11,11 +11,11 @@ def test_score():
 
 
 def test_single_thread():
-    scoring_system = SingleThread({'fakewiki': fakewiki}, timeout=1)
+    scoring_system = SingleThread({'fakewiki': fakewiki}, timeout=1.0)
```
Comment (halfak) [inline ores/scoring_systems/tests/test_single_thread.py]: Should have been a floating point number

File: ores/scoring_systems/tests/test_single_thread.py [hunk 2]
```diff
@@ -11,11 +11,11 @@ def test_score():
...
 
     response = scoring_system.score(
         ScoreRequest("fakewiki", [1], ["fake"],
-                     injection_caches={1: {wait_time: 2}}))
+                     injection_caches={1: {wait_time: 2.0}}))
```
Comment (halfak) [inline ores/scoring_systems/tests/test_single_thread.py]: Needs to be floating point

File: ores/scoring_systems/tests/util.py [hunk 3]
```diff
@@ -34,8 +35,9 @@ def score(self, feature_values):
 
 class FakeExtractor(Extractor):
 
-    def extract(self, rev_ids, features, *args, **kwargs):
-        return [(None, [0 for feature in features]) for rev_id in rev_ids]
+    def extract(self, rev_ids, features, *args, caches={}, **kwargs):
+        return [(None, list(solve(features, cache=caches.get(rev_id, {}))))
+                for rev_id in rev_ids]
```
Comment (halfak) [inline ores/scoring_systems/tests/util.py]: The old version here was doing literally nothing.  We need to it abide by the extraction pattern for the test to pass.

File: ores/scoring_context.py [hunk 4]
```diff
@@ -152,28 +151,32 @@ def extract_root_dependency_caches(
...
         for rev_id in rev_ids:
             injection_cache = injection_caches.get(rev_id, {}) \
                               if injection_caches is not None else {}
-            root_caches[rev_id] = dict(injection_cache.items())
+            _injection_caches[rev_id] = dict(injection_cache.items())
 
         # Find our root datasources
         root_datasources = \
             list(set(self._generate_root_datasources(model_names)))
+        print(root_datasources, _injection_caches)
```
Comment (Ladsgroup) [inline ores/scoring_context.py]: Leftover from debug

Comment (halfak) [reply ores/scoring_context.py]: Woops!  Good catch!

**Comment by Ladsgroup** [comment]
$ docker-compose exec ores-worker flake8 /ores
/ores/ores/scoring_context.py:6:1: F401 'revscoring.datasources.Datasource' imported but unused

**Comment by halfak** [comment]
Build failing seems to be unrelated.  Looks like our tests may be broken due to a version conflict.

**Comment by halfak** [comment]
Weird.  A strange new failure showed up.  Digging into it now.

**Pull request merged**